execution/aggregate/hashaggregate.go

```diff
@@ -140,14 +132,16 @@ func (a *aggregate) Next(ctx context.Context) ([]model.StepVector, error) {
 		}
 		a.lastBatch = nil
 	}
+
 	for {
-		next, err := a.next.Next(ctx)
+		n, err := a.next.Next(ctx, a.tempBuf)
```

Comment:
Do we want to initialize `tempBuf` to the batch size?

Reply:
Ah ok, I just saw it's initialized here: https://github.com/thanos-io/promql-engine/pull/664/changes#diff-8cffb8d32523e51056687466b98e3a24a7b27643c72bbaf0a0757ec210947e10R206